File: scripts/maui-vml-verify/corpus/c/skel.c
/* skel.c —— VML 骨架程序（C）：函数 / 数组 / 循环 / 一次 ui 调用 / 打印一行。
 *
 * 习惯用法抄自 Examples/c/tetris.c：`#include <waycoder_ui.h>`，直接调 ui_* 包装。
 *
 * 打印**有意不走 stdio**：docs/VML宿主接口.md 记着本环境 printf 的格式化路径会崩、
 * puts 输出会串行/重复，所以用共享库的 print_str / print_int / newline（SYSCALL #1/#6/#4）。
 *
 * ⚠ `__stdcall` 现在**只是留着的历史装饰**，写不写都一样：2026-09-17 调用约定统一之后，
 *   前端只认一条规则「实参全部右到左压栈、调用方清栈」，`__stdcall` / `__cdecl` /
 *   `__fastcall` 仍能解析但**不再影响代码生成**（见 docs/VML调用约定统一.md）。
 *   老注释说「声明成 cdecl 会让每条调用多释放 4 字节栈」——那说的是统一之前的形态，已作废。
 */
#include <waycoder_ui.h>

__stdcall void print_str(char* s);
__stdcall void print_int(int v);
__stdcall void newline(void);

int inc(int x) { return x + 1; }

int main(void) {
    int a[4] = {1, 2, 3, 4};
    int s = 0;
    int i;
    for (i = 0; i < 4; i++) {
        a[i] = inc(a[i]);
        s = s + a[i];
    }
    ui_rect(10, 10, 50, 50, 0xFFFF0000, 1, 0, 0);
    ui_present();
    print_str("SKEL-SUM=");
    print_int(s);
    newline();
    return 0;
}

File: scripts/maui-vml-verify/corpus/swift/skel.swift
// skel.swift —— VML 骨架程序（Swift）。
//
// 习惯用法抄自 Examples/swift/snake.swift、plane.swift（裸调 ui_*，无 import/声明）。
// 三条实测约束（snake.swift 文件头）：① 模块级标量读出来是垃圾 ⇒ 状态放数组、常量写进 main 的
// 局部变量；② 只开**一个**全局数组（多个全局数组互相踩），这里干脆全放 main 内的局部；
// ③ 颜色写负数十进制（-65536 = 0xFFFF0000）。
// print 逐参输出、**不插分隔符**、末尾只补一次换行（CodeGenerator.cs:759-781）⇒ 两个参数会拼成一行。
// 循环用 for-in（GenerateForEach，支持 0...3 闭区间）；示例里只出现过 while，这条是新写法。
func inc(x: Int) -> Int { return x + 1 }

func main() {
    var a = [1, 2, 3, 4]
    var s = 0
    for i in 0...3 {
        a[i] = inc(a[i])
        s = s + a[i]
    }
    ui_rect(10, 10, 50, 50, -65536, 1, 0, 0)
    ui_present()
    print("SKEL-SUM=", s)
}
